lib/builder_web.ml

```diff
@@ -352,6 +352,39 @@ let routes ~datadir ~cachedir ~configdir ~expired_jobs =
     let job_name = Dream.param req "job" in
     let platform = Dream.query req "platform" in
     redirect_latest req ~job_name ~platform ~artifact:""
+
+  in
+
+  let return_latest_uuid req =
+    let job_name = Dream.param req "job" in
+    let platform = Dream.query req "platform" in
+    (Dream.sql req (Model.job_id job_name) >>= Model.not_found >>= fun job_id ->
+      Dream.sql req (Model.latest_successful_build_uuid job_id platform))
+      >>= Model.not_found
+      |> if_error "Error getting job" >>= fun build ->
+        let json_response =
+          `Assoc [ "uuid", `String (Uuidm.to_string build) ] |> Yojson.Basic.to_string
+        in
+        Dream.json ~status:`OK json_response |> Lwt_result.ok
+  in
+
+  let return_uuid req =
+    Dream.query req "sha256" |> Option.to_result ~none:(`Msg "Missing sha256 query parameter")
+    |> Lwt.return
+    |> if_error ~status:`Bad_Request "Bad request" >>= fun hash_hex ->
+    begin try Ohex.decode hash_hex |> Lwt_result.return
+      with Invalid_argument e -> Lwt_result.fail (`Msg ("Bad hex: " ^ e))
+    end
+    |> if_error ~status:`Bad_Request "Bad request" >>= fun hash ->
+    Dream.sql req (Model.build_hash hash) >>= Model.not_found
+    |> if_error "Internal server error" >>= fun (job_name, build) ->
+      let json_response =
+        `Assoc [
+          "uuid", `String (Uuidm.to_string (build.Builder_db.Build.uuid));
+          "job", `String job_name
```

Comment:
Would it make sense to return more information from the build? For example, we could return the platform and execution {start,finish}, and maybe even if there's a main binary. For more information (e.g. other build artifacts) we would need to do more database queries...